merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/AutoValueMappers.java

```diff
@@ -167,4 +303,79 @@ private static CodeBlock castIfGeneric(final Element record) {
       return CodeBlock.of("(Class<$T>) (Object) ", TypeName.get(record.asType()));
     }
   }
+
+  /**
+   * Turn a given string into a valid Java identifier
+   * @param str a string that may or may not be a valid Java identifier
+   * @return a string that is a valid Java identifier
+   *
+   * Any illegal characters are replaced by `$` followed by their UTF_8 integer representation as a decimal number.
+   * This helps ensure that strings the differ only in illegal characters get unique identifiers. (e.g. "m/s" and "m*s")
+   *
+   * E.g. "<Hello (Worl%d!)" becomes "$60Hello$32$40Worl$37d$33$41"
+   */
+  public static String getIdentifier(String str) {
+    if (str.length() == 0) throw new IllegalArgumentException("Cannot turn empty string into an identifier");
+    final var strBytes = str.getBytes(StandardCharsets.UTF_8);
+    final var identifier = new StringBuilder();
+    if (Character.isJavaIdentifierStart(str.charAt(0))) {
+      identifier.append(str.charAt(0));
+    } else {
+      identifier.append('_');
```

Comment:
Minor, ignorable, the leading `_` is unnecessary on the first char in a string

Reply:
Closing as I mistakenly thought that this was generating `$INT` and not just `INT`